### DISCUSSION TITLE
Clarification on when zim & zim modules make some request to Github or any external services

### DISCUSSION THREAD

**Discussion by 2Belette** [zimfw/zimfw#515] [General]
Hi, 

I realized that when I change the theme, zim made some connections to githubusercontent.com and so I would like to see if / when external connections are made and where / for which purposes.
For example, is there any regular hearbeat, statistics, version update or so on ?

Is there a way to prevent any connections at all ?

Many thanks

**Comment by ericbn** [comment]
@2Belette, each module is downloaded from GitHub individually the first time it's added to .zimrc. Modules are downloaded from github.com by default. If you enable the degit tool for all or for individual modules, each is downloaded from api.github.com instead.

Also, zimfw will check for new versions of itself by connecting to github.com. This only happens if the last check was done for more than 30 days and only when zimfw is called. The `zimfw check`, `zimfw install`, `zimfw update`, `zimfw upgrade` and `zimfw check-version` actions also all connect to github.com when called.

Not sure about githubusercontent.com, but that also belongs to GitHub. Maybe some of the scenarios above will redirect to this host as part of the GitHub interface, not sure.

Once you've installed zimfw and all your modules, zimfw will not cause any network connections during startup. You can also set
```zsh
zstyle ':zim' disable-version-check yes
```
to disable the new version check when zimfw is called. And of course you should not call any of the zimfw actions mentioned above if you won't want to trigger network connections.

Here's also how you can leave the Zim initialization in your .zshrc if you really don't want any network connections after everything was installed:
```zsh
zstyle ':zim' disable-version-check yes
ZIM_HOME=~/.zim
# Initialize modules.
source ${ZIM_HOME}/init.zsh
```
(See details [here](https://github.com/zimfw/zimfw/blob/master/README.md#set-up-zshrc))

**Comment by 2Belette** [comment]
@ericbn never had chance to say thank you for the nice explanation !
